src/posix/platform/udp.cpp

```diff
@@ -402,19 +436,22 @@ void platformUdpInit(const char *aIfName)
         DieNow(OT_EXIT_INVALID_ARGUMENTS);
     }
 
-    sPlatNetifIndex = if_nametoindex(aIfName);
-
-    if (sPlatNetifIndex == 0)
+    if (aIfName != gNetifName)
     {
-        perror("if_nametoindex");
+        VerifyOrDie(strlen(aIfName) < sizeof(gNetifName) - 1, OT_EXIT_INVALID_ARGUMENTS);
+        strcpy(gNetifName, aIfName);
+        gNetifIndex = if_nametoindex(gNetifName);
+        VerifyOrDie(gNetifIndex != 0, OT_EXIT_ERROR_ERRNO);
     }
+
+    assert(gNetifIndex != 0);
 }
 
 void platformUdpProcess(otInstance *aInstance, const fd_set *aReadFdSet)
 {
     otMessageSettings msgSettings = {false, OT_MESSAGE_PRIORITY_NORMAL};
 
-    VerifyOrExit(sPlatNetifIndex != 0, OT_NOOP);
+    VerifyOrExit(gNetifIndex != 0, OT_NOOP);
```

Comment:
if `gNetifIndex == 0`, the program should have exited in `platformUdpInit`, 
Is it still necessary to check here?

Reply:
```suggestion
```